single_tech_samples/datafactory/sample3_data_pre_processing_with_azure_batch/README.md

```diff
@@ -0,0 +1,195 @@
+
+![use-case](images/pre-processing-usecase.map.drawio.svg)
+
+Details about [how to run the pipeline](#24-deployment-validation-and-execution) can be found in the later sections of this document.
+
+### 1.3. Architecture
+
+The below diagram illustrates the high level design showing the ADF and the azure batch integration for running container workloads on azure batch:
+
+![alt text](images/adf-batch-integration-design.svg "Design Diagram")
+
+#### **Architecture Design Components**
+
+- **Raw Zone:** This is an ADLS account where the ingested data lands.
+
+- **Extracted Zone:** This is an ADLS account where the extracted data will be stored.
+
+- **ADF Pipeline:** This is an ADF pipeline which can have a scheduled trigger to pick data from raw zone and sends it to azure batch for extraction.
+
+- **Azure Batch:**  It will extract the bag file data and stored the extracted contents to extracted zone. Azure Batch has following components:
+
+  - **Orchestrator Pool:** This is a batch pool which runs a batch application, see [sample here](src/orchestrator-app/README.md) on an ubuntu node. ADF pipeline uses a custom activity to invoke this application. This application acts as an orchestrator for creating jobs and tasks and monitoring those for completion on execution pool described below.
+
+  - **Execution Pool:** This is a a batch pool which is responsible for running container work loads. This pool has start up task configuration where it mounts the raw and extracted ADLS zones as NFS mounts to its nodes and when containers are spinned to process the actual workloads and the same mounts are mounted to containers as well via container run options. Orchestrator pool creates jobs and tasks for execution pool.
+
+      ADLS storage mounts on execution pool nodes and containers will help containers to process large files without downloading the locally.
+
+- **Application Insights:** It monitors the health of execution and orchestrator pool nodes. We can integrate the application insights with azure batch application and the container images for end to end tracing of jobs.
+
+- **Azure Container Registry(ACR):**  In this sample execution pool is configured with a azure container registry to pull container images, but we can have any container registry configured.
+
+- **Managed Identity:** This is a user assigned managed identity which will be assigned to both of the batch pools and batch pool will use it to access any other required resources like container registry, keyvault, storage account etc.
+
+### 1.4. Technologies used
+
+The following technologies are used to build this sample:
+
+- [Azure Data Factory](https://azure.microsoft.com/en-in/products/data-factory/)
```

Comment:
Remove locale information from URLs:
https://azure.microsoft.com/en-in/products/data-factory/
=>
https://azure.microsoft.com/products/data-factory/